source/state_representation/src/Robot/JointState.cpp

```diff
@@ -45,6 +45,36 @@ void JointState::set_zero() {
   this->torques.setZero();
 }
 
+JointState JointState::Zero(const std::string& robot_name, unsigned int nb_joints) {
+  JointState zero = JointState(robot_name, nb_joints);
+  // as opposed to the constructor specify this state to be filled
+  zero.set_filled();
+  return zero;
+}
+
+JointState JointState::Zero(const std::string& robot_name, const std::vector<std::string>& joint_names) {
+  JointState zero = JointState(robot_name, joint_names);
+  // as opposed to the constructor specify this state to be filled
+  zero.set_filled();
+  return zero;
+}
+
+JointState JointState::Random(const std::string& robot_name, unsigned int nb_joints) {
+  JointState random = JointState(robot_name, nb_joints);
+  // set all the state variables to random
+  random.set_state_variable(Eigen::VectorXd::Random(random.get_size() * 4),
```

Comment:
why x4? Unit distribution or `M_PI` for revolute joints would seem a bit more justified

Reply:
This `*4` is for the size not the actual random distribution as we have to fill `positions`, `velocities`, `accelerations` and `torques`. If we want to fill a specific state variables then we should use the `random` of the other classes.

Reply:
Thank you, I see now!